### ISSUE TITLE
Parse number input value as JS number

### ISSUE THREAD

**Issue by lucafaggianelli** [dohomi/react-hook-form-mui#117]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Summary 💡

The `<TextFieldElement>` returns a string value even for numeric fields (i.e. with `type="number"`), it would be nice to have an option or a separate component (that is `<NumberFieldElement>`) to return the value as a JS Number.

## Why?

I'm using a setup including RHF + RHF-MUI + Prisma as DB ORM on the backend and Prisma doesn't accept strings where the input should e number, now it would be easy to parse a single variable, but my DB schema is pretty complex and dynamic (I use MongoDB) so parsing strings to int would be a pain.

### Examples 🌈

I ended up creating a custom element, copy-pasting the original TextElementField and changing the `onChange` method but it would be nice to include it in the js package:

```tsx
import { TextField, TextFieldProps } from '@mui/material';
import {
  Control,
  Controller,
  ControllerProps,
  FieldError,
  Path,
} from 'react-hook-form';
import { FieldValues } from 'react-hook-form/dist/types/fields';

export type NumberFieldElementProps<T extends FieldValues = FieldValues> = Omit<
  TextFieldProps,
  'name'
> & {
  validation?: ControllerProps['rules'];
  name: Path<T>;
  parseError?: (error: FieldError) => string;
  control?: Control<T>;
};

export default function NumberFieldElement<
  TFieldValues extends FieldValues = FieldValues,
>({
  validation = {},
  parseError,
  type,
  required,
  name,
  control,
  ...rest
}: NumberFieldElementProps<TFieldValues>): JSX.Element {
  if (required && !validation.required) {
    validation.required = 'This field is required';
  }

  if (type === 'email' && !validation.pattern) {
    validation.pattern = {
      value:
        // eslint-disable-next-line no-useless-escape
        /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
      message: 'Please enter a valid email address',
    };
  }

  return (
    <Controller
      name={name}
      control={control}
      rules={validation}
      render={({
        field: { value, onChange, onBlur, ref },
        fieldState: { invalid, error },
      }) => (
        <TextField
          {...rest}
          name={name}
          value={value ?? ''}
          onChange={(ev) => {
            onChange(+ev.target.value);
            if (typeof rest.onChange === 'function') {
              rest.onChange(ev);
            }
          }}
          onBlur={onBlur}
          required={required}
          type={type}
          error={invalid}
          helperText={
            error
              ? typeof parseError === 'function'
                ? parseError(error)
                : error.message
              : rest.helperText
          }
          inputRef={ref}
        />
      )}
    />
  );
}
```

**Comment by dohomi** [comment]
@lucafaggianelli do you want to add your solution as a PR? Currently a `type=number` prop is available at the property level and there it could parse the value to a number. That would supersede a new component.

**Comment by jamiedruce** [comment]
Came here to make the same suggestion, but was beaten by one day, ha!

**Comment by dohomi** [comment]
I released v5.12.4 to resolve this issue

**Comment by lucafaggianelli** [comment]
Ah wow thai Washington fast! Thank you
